pkg/metallb/install.go

```diff
@@ -0,0 +1,131 @@
+package metallb
+
+import (
+	_ "embed"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/ns"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+var (
+	//go:embed yaml/metallb-operator.yaml
+	metallbOperator string
+
+	//go:embed yaml/metallb.yaml
+	metallb string
+)
+
+func InstallIfNotExist(t test.TestHelper, kubeConfigs ...string) {
+	if len(kubeConfigs) == 0 {
+		installWithOC(t, *oc.DefaultOC)
+		return
+	}
+
+	for _, config := range kubeConfigs {
+		oc := oc.WithKubeconfig(config)
+		if oc == nil {
+			t.Errorf("failed to create oc config from kubeconfig file: %s", config)
+		}
+		installWithOC(t, *oc)
+	}
+}
+
+func installWithOC(t test.TestHelper, oc oc.OC) {
+	if checkIfMetalLbOperatorExists(t) {
+		t.Log("MetalLB operator already exists - skip installation of the operator")
+	} else {
+		installOperator(t, oc)
+	}
+	if checkIfMetalLbControllerExists(t) {
+		t.Log("MetalLB controller already exists - skip deploying MetalLB")
+	} else {
+		deployMetalLB(t, oc)
+	}
+	if checkIfIPAddressPoolExists(t) {
+		t.Log("IPAddressPool already exists - skip applying IPAddressPool")
+	} else {
+		createAddressPool(t, oc)
+	}
+}
+
+func installOperator(t test.TestHelper, oc oc.OC) {
+	t.Log("Install MetalLB operator")
+	oc.ApplyString(t, ns.MetalLB, metallbOperator)
+	retry.UntilSuccess(t, func(t test.TestHelper) {
+		// pattern "cmd || true" is used to avoid getting fatal error
+		shell.Execute(t, fmt.Sprintf("oc get deployments -n %s metallb-operator-controller-manager || true", ns.MetalLB),
```

Comment:
This won't work properly when the `oc` passed to this function is not the default `oc`. 

And this is exactly why I said in the other PR that using `shell.Execute` to run a custom `oc` command is not okay.

Reply:
Yeah... I didn't find this, because I was testing missing operator, MetalLB and IPAddressPool on the cluster for which default kubeconfig is used.

Reply:
I will have to pass kubeconfigs to each of these commands.

Reply:
How about creating `oc.ResourceExists(t, ns, kind, name)`? Seems like we need this in several places.

Reply:
Done